addon/components/freestyle-dynamic-input.js

```diff
@@ -0,0 +1,28 @@
+import Ember from 'ember';
+import layout from '../templates/components/freestyle-dynamic-input';
+
+const { computed, computed: { equal } } = Ember;
+
+export default Ember.Component.extend({
+  layout,
+  classNames: ['FreestyleDynamic-input'],
+
+  isCheckbox: equal('inputType', 'checkbox'),
+  isTextarea: equal('inputType', 'textarea'),
+  isSelect: equal('inputType', 'select'),
+
```

Comment:
Would be very useful to have `inputType="number"` and parse it as a real integer.